Lib/unittest/mock.py

```diff
@@ -1904,6 +1904,8 @@ def method(self, /, *args, **kw):
     '__str__': lambda self: object.__str__(self),
     '__sizeof__': lambda self: object.__sizeof__(self),
     '__fspath__': lambda self: f"{type(self).__name__}/{self._extract_mock_name()}/{id(self)}",
+    '__aenter__': lambda self: AsyncMockMixin._mock_call(self),
```

Comment:
`AsyncMockMixin._mock_call` is an async def so I guess it should be awaited. There are some warnings but not sure if they are related to this.

```
./python.exe -m unittest Lib.unittest.test.testmock.testasync
........../Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/mock.py:740: RuntimeWarning: coroutine 'AsyncMockMixin._mock_call' was never awaited
  return object.__setattr__(self, name, value)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
...../Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/mock.py:2084: RuntimeWarning: coroutine 'AsyncMockMixin._mock_call' was never awaited
  self.name = name
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
.....................................
----------------------------------------------------------------------
Ran 52 tests in 0.508s

OK
```